regression: unused import causing downstream typecheck failure

**What happened?**
After upgrading to 2.11.0, my build started failing with:
```
> tsc --noEmit
node_modules/@octokit/types/src/RequestInterface.ts:2:1 - error TS6133: 'EndpointOptions' is declared but its value is never read.
2 import { EndpointOptions } from "./EndpointOptions";
  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**What did you expect to happen?**
Pass type checking when imported from my project, even if my `tsconfig.json` is strict.

**What the problem might be**
My `tsconfig.json` file has the following 4 strict options turned on: 
- noUnusedLocals
- noUnusedParameters
- noImplicitReturns
- noFallthroughCasesInSwitch

`@octokit/types` is published as `.ts` files, which causes my typescript to check it (using *my* `tsconfig.json`) when imported via `node_modules`.

Workaround: remove unused import and republish.
More permanent solutions (to avoid similar issues):
- add a `tsconfig.json` to repo with strict options all turned on. this way, you'd know code is compatible.
- build package using tsc and publish the .d.ts files.

--- a/package.json
+++ b/package.json
@@ -14,7 +14,7 @@
     "lint": "prettier --check '{src,test,scripts}/**/*.{js,ts,json}' README.md package.json !src/generated/* !scripts/update-endpoints/generated/*",
     "lint:fix": "prettier --write '{src,test,scripts}/**/*.{js,ts,json}' README.md package.json !src/generated/* !scripts/update-endpoints/generated/*",
     "pretest": "npm run -s lint",
-    "test": "npx tsc --noEmit --declaration src/index.ts test.ts",
+    "test": "npx tsc --noEmit --declaration --noUnusedLocals src/index.ts test.ts",
     "update-endpoints": "npm-run-all update-endpoints:*",
     "update-endpoints:fetch-json": "node scripts/update-endpoints/fetch-json",
     "update-endpoints:typescript": "node scripts/update-endpoints/typescript"

--- a/src/RequestInterface.ts
+++ b/src/RequestInterface.ts
@@ -1,5 +1,4 @@
 import { EndpointInterface } from "./EndpointInterface";
-import { EndpointOptions } from "./EndpointOptions";
 import { OctokitResponse } from "./OctokitResponse";
 import { RequestParameters } from "./RequestParameters";
 import { Route } from "./Route";

